core/l10n.js

```diff
@@ -59,8 +59,10 @@ define('core/l10n',
 
     return {
         getDirection: function(context) {
+            // Try to get browser language, if undefined use default locale
+            var browserLanguage = window.navigator.language || 'en-US';
             var language = context ? context.language :
-                                     window.navigator.l10n.language;
+                                     browserLanguage;
```

Comment:
i think the *insertlocalehere*.js which gets injected should be some javascript code which populates the window.navigator.l10n object. The content which gets returned from the request is no js content, furthermore some generated html content

```html
<!--This is a generated file from dev.html.-->
<!--Read marketplace-frontend.readthedocs.org for more information.-->

<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1, maximum-scale=1">
    <meta property="og:title" content="Firefox Marketplace">
    <meta property="og:type" content="website">
    <meta property="og:image" content="/media/img/logos/128.png">
    <meta name="theme-color" content="#fff">

    <title>Firefox Marketplace</title>

    <link rel="icon" href="/media/img/logos/128.png" sizes="128x128">
    <link rel="icon" href="/media/img/logos/64.png" sizes="64x64">
    <link rel="icon" href="/media/img/logos/32.png" sizes="32x32">

    
      <link rel="stylesheet" href="/media/css/splash.styl.css">
      <!-- These styles need to come first. -->
      <link rel="stylesheet" href="/media/css/reset.styl.css">
      <link rel="stylesheet" href="/media/css/typography.styl.css">
      <link rel="stylesheet" href="/media/css/site.styl.css">
      <link rel="stylesheet" href="/media/css/footer.styl.css">
      <link rel="stylesheet" href="/media/css/header.styl.css">
      <link rel="stylesheet" href="/media/css/navbar.styl.css">
      <link rel="stylesheet" href="/media/css/category-icons.styl.css">
      <link rel="stylesheet" href="/media/css/tiles.styl.css">
      <link rel="stylesheet" href="/media/css/infobox.styl.css">
      <!-- These styles don't need to be in an order. -->
      <link rel="stylesheet" href="/media/css/account.styl.css">
      <link rel="stylesheet" href="/media/css/app-list.styl.css">
      <link rel="stylesheet" href="/media/css/app-list-filters.styl.css">
      <link rel="stylesheet" href="/media/css/buttons.styl.css">
      <link rel="stylesheet" href="/media/css/buttons-loadmore.styl.css">
      <link rel="stylesheet" href="/media/css/categories.styl.css">
      <link rel="stylesheet" href="/media/css/category-dropdown-desktop.styl.css">
      <link rel="stylesheet" href="/media/css/compat-filter.styl.css">
      <link rel="stylesheet" href="/media/css/debug.styl.css">
      <link rel="stylesheet" href="/media/css/desktop-promo.styl.css">
      <link rel="stylesheet" href="/media/css/detail/base.styl.css">
      <link rel="stylesheet" href="/media/css/detail/content-ratings.styl.css">
      <link rel="stylesheet" href="/media/css/detail/header.styl.css">
      <link rel="stylesheet" href="/media/css/detail/more-info.styl.css">
      <link rel="stylesheet" href="/media/css/detail/info.styl.css">
      <link rel="stylesheet" href="/media/css/detail/post-install-message.styl.css">
      <link rel="stylesheet" href="/media/css/detail/reviews.styl.css">
      <link rel="stylesheet" href="/media/css/detail/upsell.styl.css">
      <link rel="stylesheet" href="/media/css/elements--nav.styl.css">
      <link rel="stylesheet" href="/media/css/elements--select.styl.css">
      <link rel="stylesheet" href="/media/css/error-screens.styl.css">
      <link rel="stylesheet" href="/media/css/feed/feed.styl.css">
      <link rel="stylesheet" href="/media/css/feed/brands.styl.css">
      <link rel="stylesheet" href="/media/css/feed/collections.styl.css">
      <link rel="stylesheet" href="/media/css/feed/colors.styl.css">
      <link rel="stylesheet" href="/media/css/feed/curves.styl.css">
      <link rel="stylesheet" href="/media/css/feed/icon-grid.styl.css">
      <link rel="stylesheet" href="/media/css/feed/landing.styl.css">
      <link rel="stylesheet" href="/media/css/feed/layout.styl.css">
      <link rel="stylesheet" href="/media/css/feed/shelves.styl.css">
      <link rel="stylesheet" href="/media/css/feedback.styl.css">
      <link rel="stylesheet" href="/media/css/forms.styl.css">
      <link rel="stylesheet" href="/media/css/lightbox.styl.css">
      <link rel="stylesheet" href="/media/css/image-deferrer.styl.css">
      <link rel="stylesheet" href="/media/css/modal.styl.css">
      <link rel="stylesheet" href="/media/css/newsletter.styl.css">
      <link rel="stylesheet" href="/media/css/notification.styl.css">
      <link rel="stylesheet" href="/media/css/promo-grid.styl.css">
      <link rel="stylesheet" href="/media/css/pretty-select.styl.css">
      <link rel="stylesheet" href="/media/css/previews.styl.css">
      <link rel="stylesheet" href="/media/css/reviews.styl.css">
      <link rel="stylesheet" href="/media/css/search.styl.css">
      <link rel="stylesheet" href="/media/css/settings-hover-desktop.styl.css">
      <link rel="stylesheet" href="/media/css/standalone.styl.css">
      <link rel="stylesheet" href="/media/css/view-all-tab.styl.css">

      <link rel="stylesheet" href="/media/css/marketplace-elements.styl.css">
    

    <!-- TODO: Add all these tags to Zamboni's commonplace `index.html` -->
    <!-- New W3C-standardised manifest with application metadata -->
    <link rel="manifest" href="manifests/dev.json">
    <!-- Fallback application metadata for legacy browsers -->
    <meta name="application-name" content="Firefox Marketplace">
    <link rel="icon" sizes="128x128" href="/media/img/app-icons/128.png">
    <!-- Add to homescreen for Chrome on Android -->
    <meta name="mobile-web-app-capable" content="yes">
    <link rel="icon" sizes="128x128" href="/media/img/app-icons/128.png">
    <!-- Add to homescreen for Safari on iOS -->
    <meta name="apple-mobile-web-app-title" content="Firefox Marketplace">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" href="/media/img/app-icons/128.png">
    <!-- Tile icon for Win8 (icon should be 144x144) -->
    <meta name="msapplication-TileImage" content="/media/img/app-icons/128.png">
    <meta name="msapplication-TileColor" content="#fff">
  </head>

  <body class="overlayed" data-media="" data-local-html>
    <noscript><p>Sorry, you need JavaScript to access the Marketplace.</p></noscript>
    <div id="splash-overlay">
      <div class="throbber"><b></b><b></b><b></b></div>
    </div>

    <header id="site-header" class="header site-header"></header>

    <div class="cloak"></div>
    <main>
      <nav id="site-nav" class="site-nav" role="navigation"></nav>
      <div id="page" class="page" role="main"></div>
    </main>

    <footer id="site-footer"></footer>

    <!-- Screenshot lightbox. -->
    <div id="lightbox">
      <section>
        <ul class="content"></ul>
        <a class="close" href="#">&times;</a>
      </section>
    </div>

    
      <script data-main="/media/js/main" src="/media/js/lib/require.js"></script>
    
  <script src="http://localhost:9675/livereload.js"></script>
</body>
</html>
```

this is what the request of `/media/locales/de.js` returns which results in the following message `SyntaxError: expected expression, got '<'`